src/main/shell-session/shell-session.ts

```diff
@@ -358,6 +361,8 @@ export abstract class ShellSession {
       .filter(Boolean)
       .join();
 
+    this.shellEnvModifiers.map(modifier => env = { ...env, ...modifier({ context: this.cluster.contextName,  kubeconfigPath, id: this.cluster.id }, env) });
```

Comment:
Probably not what we want to do if multiple extensions modify the env. Not sure what we should do. Allow user to enable/disable some? Select the order they are applied?

Reply:
This should be a reduce call instead. I think leaving the order be undefined is fine.

Something like:
```
const env = clearKubeconfigEnvVars(JSON.parse(JSON.stringify(await shellEnv())));
...
const ctx = { ... };
const modifiedEnv = this.shellEnvModifiers.reduce((env, modifier) => modifier(ctx, env), env);
```

Reply:
>  I think leaving the order be undefined is fine.

Really? The order can vary from time to time, right (across restarts)? Extensions might clobber each other's vars/values.

Reply:
I also wonder if it's a good idea to offer the whole default shell env to the extension, to be freely modified.

Reply:
That is true, especially if this is meant to be seamless for the user. Otherwise, we could offer the user a list of Shell Envs in the dock UI and let them decide how they are applied when opening a terminal. This of course assumes the user knows what they're doing.

Reply:
That is true (about users). I think we should start with the order unspecified and if absolutely necessary add in some mechanism in the future.